### ISSUE TITLE
Why activeProbes are defined at class level ?

### ISSUE THREAD

**Issue by SergeStinckwich** [cormas/cormas#58]
Why ```activeProbes``` is defined at the class level in the class ``CormasModel`` ? This means, if we launch 2 simulations of the same model, they have the same activeProbes.

I think activeProbes should move at the instance level of ``CormasModel``.

**Comment by pierbo** [comment]
Well, this is a problem. Class variables are useful to acces a given value during a simulation. For example, to know the current time step, any entity can read this value by asking `MyModel timeStep`. Here `timeStep` is a class var. If we run several simulations simultaneously, `timeStep` may be different for each sim. So, it doesn't work. 
The decision we took was to run ONE simulation at a time. For parallelisation, we should use several images.
Adding an attribute (`myScheduler` for example), could be a solution. But we don't want that any entity can access the instance of its CormasModel by default. Because, through this association, the Entities designed by modelers have access to every thing. If the modeler needs that, it should be specified at the design stage of their model.
   Pierre

**Comment by SergeStinckwich** [comment]
I don’t see the problem. Instead of doing ‘’MyModel timeStep’’ you do: ‘’myModel timeStep’’ where myModel is a référence to the model. So in this case, this is better because you can have a distinct timeStep for each simulations.

**Comment by pierbo** [comment]
No, this is not the case: the instances of Entity have no access to cormasModel.
By experience, we know that the modelers will, through this association (Entity -> cormasModel), access to any kind of entity (instead of specifying specific associations in their model). There is no attribute  cormasModel (or myScheduler) by default. A modeler can decide to do that, but it is not done by default.

**Comment by SergeStinckwich** [comment]
You can access if you want because there is only one instance of your model. You can do: MyModel uniqueInstance and that’s it :-) if you don’t say to modelers that there is a myModel instance, they will not try to access it. I don’t see why you are afraid here.

**Comment by ChLePage** [comment]
This is an interesting point, indeed... As explained by Pierre, the point is to force modellers to make explicitly visible all the associations at the entity class level. Maybe we could think about a way to specify, for each entity, "accessors" to myModel public accessors. This could be done through a dedicated interface, similar to the one used to add "attributes"? Or we could add this possibility directly in the "attributes" interface ?

**Comment by ChLePage** [comment]
oups! I did not want to close the issue, sorry! I have to become more familiar with this stuff... ;-))

**Comment by nbecu** [comment]
Hi,
Indeed, the idea is that, when requiered, the modeler is forced to specify the asociations between the entities (or between an entity and the cormasModel) so that he understands the notion of autonomous agent (by default an autonomous agent doesn't know anyone else than himself).

Yet, there is trick !
At the Entity level the method 'currentSimulation' simulation exists . it returns -> ^self class modelClass currentSimulation
So, some lazy modelers, like me, sometime  access to other entities from an Entity level using an instruction like -> self currentSimulation theCells at:2
Conceptually it shouldn't be done, but yet, it is possible !

**Comment by pierbo** [comment]
Hola desde Montevideo !
Just one last think about class var : many models use class var, for ex to know the T°, the current day, the price of a crop, etc. These var may change during a simulation (market fluctuation, CC,...). So, these values may be different from one simu to another. 
So, this point implies that we cannot run several simulations in parallel on the same image. To launch several simu in parallel, we are obliged not only to have several processes, but especially to launch several images. Or we have to forbid modelers to use class var, but I don't like it...
